Tests/Security/Core/User/EntityUserProviderTest.php

```diff
@@ -57,6 +57,7 @@ public function testLoadUserByUsernameThrowsExceptionWhenUserIsNull()
 
             $this->fail('Failed asserting exception');
         } catch (\RuntimeException $e) {
+            \assert($e instanceof UsernameNotFoundException);
```

Comment:
This sounds wrong.

Reply:
changed it as it seem you dislike using `assert` to help static analyzers understand the types in certain cases